### PR TITLE
feat(algoliaAgent): track Next.js version as Algolia agent

### PR DESCRIPTION
**Pull request by FabienMotte**
**Summary**

This PR tracks the Next.js version by adding it as an Algolia agent when a Next.js app is detected.
To get the Next.js version:
- on the browser: `window.next.version` (e.g. `12.1.6`).
- in SSR: `process.env.NEXT_RUNTIME` (`nodejs` or `edge`).

### CODE REVIEW

**Review by Haroenv — CHANGES_REQUESTED** [review]
`npm_package_dependencies_next` is removed in newer versions of npm unfortunately, I think an incomplete picture probably is worse than no picture

File: packages/react-instantsearch-hooks/src/lib/useInstantSearchApi.ts [hunk 1]
```diff
@@ -262,3 +268,15 @@ You can ignore this warning if you are using a custom router that suits your nee
...
+
+/**
+ * Gets the version of Next.js if it is available in the `window` object,
+ * otherwise it returns the version from the npm package dependencies (in SSR).
+ */
+function getNextVersion() {
+  return (
+    (typeof window !== 'undefined' &&
+      ((window as any).next?.version as string | undefined)) ||
+    process.env.npm_package_dependencies_next
```
Comment (Haroenv) [inline packages/react-instantsearch-hooks/src/lib/useInstantSearchApi.ts:280]: process.env isn't save to access either I think, would need `?.`

**Comment by FabienMotte** [comment]
> 
Could we rely on `process.env.NEXT_RUNTIME` instead (only available from v12.1.6)?

File: packages/react-instantsearch-hooks/src/lib/useInstantSearchApi.ts (hunk 1 above)
Comment (aymeric-giraudet) [reply packages/react-instantsearch-hooks/src/lib/useInstantSearchApi.ts]: Is `process` even safe ? I guess it gets transformed by most bundlers but if using UMD this fails :/

File: packages/react-instantsearch-hooks/src/lib/useInstantSearchApi.ts [hunk 2]
```diff
@@ -177,6 +180,7 @@ export function useInstantSearchApi<TUiState extends UiState, TRouteState>(
   const cleanupTimerRef = useRef<ReturnType<typeof setTimeout> | null>(null);
   const store = useSyncExternalStore<InstantSearch<TUiState, TRouteState>>(
     useCallback(() => {
+      // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
```
Comment (Haroenv) [inline packages/react-instantsearch-hooks/src/lib/useInstantSearchApi.ts:183]: weird, why was this needed now?

**Review by Haroenv — COMMENTED** [review]
code and test looks correct, just not sure about the eslint comments

File: packages/react-instantsearch-hooks/src/lib/useInstantSearchApi.ts (hunk 2 above)
Comment (FabienMotte) [reply packages/react-instantsearch-hooks/src/lib/useInstantSearchApi.ts:183]: Yeah I tried to align my TS version on the CI one but I still have the eslint error, I'm not sure why

Comment (Haroenv) [reply packages/react-instantsearch-hooks/src/lib/useInstantSearchApi.ts:183]: I guess as it doesn't happen on CI, let's leave it as-is :)

Comment (FabienMotte) [reply packages/react-instantsearch-hooks/src/lib/useInstantSearchApi.ts]: Thanks!

**Review by Haroenv — APPROVED** [review]
